### PR TITLE
Add organization creation page

### PR DESCRIPTION
**Pull request by aa0501**
- Added screen to add new Organization
- Improved some routing

PS: Made on top of #24 (to be merged after that)

### CODE REVIEW

File: src/components/loginform/loginform.js [hunk 1]
```diff
@@ -1,106 +1,149 @@
...
-              placeholder="abc@gmail.com"
+              name="email"
             />
+            {this.state.emailError && (
+              <h6 className="form-field-error">{this.state.emailError}</h6>
+            )}
           </Form.Group>
           <Form.Group controlId="formBasicPassword">
             <Form.Label>
               <div className="formLabel">Password</div>
```
Comment (lazycipher) [inline src/components/loginform/loginform.js]: `className="formLabel"`  Maybe use a single naming convention?
https://www.freecodecamp.org/news/css-naming-conventions-that-will-save-you-hours-of-debugging-35cea737d849/

File: src/components/signupform/signupform.js [hunk 2]
```diff
@@ -1,147 +1,319 @@
...
-      case "email":
-        this.setState({ email: event.target.value });
+  onFieldChange = (e) => {
+    const { name, value } = e.target;
+    this.setState({ [name]: value }, () => {
+      this.checkFieldValidation(name);
+    });
+  };
+
+  checkFieldValidation = (field) => {
```
Comment (lazycipher) [inline src/components/signupform/signupform.js:36]: How about making `checkFieldValidation(field)` a module and use wherever we need?

File: src/reducers/authSlice.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
+import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
+const axios = require("axios");
+
+export const login = createAsyncThunk(
+  'auth/login', 
+  async (loginData, { rejectWithValue }) => {
+  const response = await axios
+    .post(
+      "http://localhost:8000/graphql", //update
```
Comment (lazycipher) [inline src/reducers/authSlice.js]: How about providing this from react env?
If we will have to change the URL, we'll have to make changes at multiple places unnecessarily.

File: src/reducers/authSlice.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        },
+      }
+    )
+    .catch((error) => {
+      if (error.response) {
+        return error.response.data.errors[0].message;
+      }
+    });
+  if(response.data != undefined) {
+    localStorage.setItem("token", response.data.data.login.token);
```
Comment (lazycipher) [inline src/reducers/authSlice.js:35]: Are we still using `localStorage` to store the auth token?

**Review by lazycipher — CHANGES_REQUESTED** [review]
@aayush-05, Please check my comments.
Rest looks good to me!

File: src/components/loginform/loginform.js (hunk 1 above)
Comment (aa0501) [reply src/components/loginform/loginform.js]: Fixed. Going with hyphenated ones

File: src/components/signupform/signupform.js (hunk 2 above)
Comment (aa0501) [reply src/components/signupform/signupform.js:36]: Good idea; but same fields are having different validations for different models. For example, somewhere shortDescription is required to be only 3 characters long, but somewhere else it has to be atleast 5 characters

File: src/reducers/authSlice.js (hunk 3 above)
Comment (aa0501) [reply src/reducers/authSlice.js]: Yeah! This has to be updated. Idk why but .env variables aren't working right now. Added //update comment anywhere it needs to be done; will be fixed afterwards

File: src/reducers/authSlice.js (hunk 4 above)
Comment (aa0501) [reply src/reducers/authSlice.js:35]: Going with this for now. Will maybe move to Cookies once the basic infrastructure gets in place

File: src/reducers/authSlice.js (hunk 3 above)
Comment (lazycipher) [reply src/reducers/authSlice.js]: What's happening with .env?
Keep in mind that we need to have the prefix `REACT_APP_` in any variable to be able to use in react!

File: src/components/signupform/signupform.js (hunk 2 above)
Comment (lazycipher) [reply src/components/signupform/signupform.js:36]: I would say standardize as much as you can if possible!
@vaibhavdaren, what do you think?

Comment (aa0501) [reply src/components/signupform/signupform.js:36]: Okay, I can think of prefixed names to differentiate it. But one query, when calling the moduralised function we will have to bind "this" reference to it first for every component which will eventually return a new function and we will calling that new function for that component. So, if the bind() method will end up creating a new function anyway; are we gaining something by moduralising this function?

File: src/reducers/authSlice.js (hunk 3 above)
Comment (aa0501) [reply src/reducers/authSlice.js]: This was a life saviour! Thanks

File: src/components/signupform/signupform.js (hunk 2 above)
Comment (aa0501) [reply src/components/signupform/signupform.js:36]: Resolved in #26

**Comment by aa0501** [comment]
Closing this. Changes included in #26

**Pull request closed without merging**